File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        ArrayList<ArrayList<Integer>> size = Arrays.stream(sizes)
                .map(row -> IntStream.of(row)
                                     .boxed()
                                     .collect(Collectors.toCollection(ArrayList::new)))
                .collect(Collectors.toCollection(ArrayList::new));
        int max1 = 0;
        int max2 = 0;
        for (int i = 0; i < size.size(); i++) {
            if (size.get(i).get(0) < size.get(i).get(1)) {
                Integer temp = size.get(i).get(0);
                size.get(i).set(0, size.get(i).get(1));
                size.get(i).set(1, temp);
            }
            if (max1 < size.get(i).get(0)) {
                max1 = size.get(i).get(0);
            }
            if (max2 < size.get(i).get(1)) {
                max2 = size.get(i).get(1);
            }
        }
        answer = max1*max2;
        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    xmax = 0
    ymax = 0
    for i in sizes:
        if i[0] < i[1]:
            temp = i[0]
            i[0] = i[1]
            i[1] = temp
        if i[0] > xmax:
            xmax = i[0]
        if i[1] > ymax:
            ymax = i[1]
    return xmax * ymax